### PR TITLE
rpm: have pybind RPMs provide/obsolete their python2 predecessors

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: http://tracker.ceph.com/issues/40099

### CODE REVIEW

**Comment by b-ranto** [comment]
Just an idea: What if we used the the `python2` bcond for this? We already have a list of python3-only/default systems listed when defining the `python2` bcond. We would only need to maintain the list for this, there.

**Comment by smithfarm** [comment]
@b-ranto Are you proposing to replace all instances of `%if 0%{?suse_version}` in this PR with `%if 0%{without python2}`?

**Comment by b-ranto** [comment]
@smithfarm Yep, that was my idea there.

**Comment by smithfarm** [comment]
I don't see any problem with @b-ranto's suggestion. @tchaikov, @tserong ?

**Comment by tserong** [comment]
Good idea, fine by me!

**Comment by smithfarm** [comment]
OK, I changed it as per @b-ranto's request.

File: ceph.spec.in [hunk 1]
```diff
@@ -733,6 +733,10 @@ Group:		Development/Libraries/Python
 Requires:	librgw2 = %{_epoch_prefix}%{version}-%{release}
 Requires:	python%{python3_pkgversion}-rados = %{_epoch_prefix}%{version}-%{release}
 Provides: 	python3-rgw = %{_epoch_prefix}%{version}-%{release}
+%if 0%{without python2}
+Provides:	python-rgw = %{_epoch_prefix}%{version}-%{release}
```
Comment (tchaikov) [inline ceph.spec.in]: i think this actually complies https://fedoraproject.org/wiki/User:Churchyard/Packaging:Python#The_.25python_provide_macro . if the "system version of python" of the targeting distro is python3, we should do this. i guess this applies to "0%{?fedora} >= 29 || 0%{?suse_version} >= 1500 || 0%{?rhel} >= 8".

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-06-06_01:54:32-rados-wip-kefu-testing-2019-06-06-0047-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-06-08_09:31:11-rados-wip-kefu-testing-2019-06-06-0047-distro-basic-smithi/#

**Pull request merged**